Sources/Utility/ArgumentParser.swift

```diff
@@ -278,8 +265,8 @@ final class AnyArgument: ArgumentProtocol, CustomStringConvertible {
     }
 
     /// Initializer for array arguments.
-    init<T>(_ arg: OptionArgument<[T]>) {
-        self.kind = T.self as! ArgumentKind.Type
+    init<T: ArgumentProtocol>(_ arg: T) where T.ArgumentKindTy: Sequence {
```

Comment:
Ok nope, scrap that again. `[ArgumentKind]` is not a concrete type and it will not select this init for `[String]` for example.